src/vmm_mad/remotes/lib/lxd/mapper/qcow2.rb

```diff
@@ -67,6 +65,21 @@ def do_unmap(device, _one_vm, _disk, _directory)
 
     private
 
+    # Detects Max number of block devices
+    def load_nbds_max
+        cmd = '/sbin/modprobe -c'
+        rc, mods_conf, e = Command.execute(cmd, false)
+
+        if !rc.zero?
+            OpenNebula.log_error("Cannot load max number of nbd devices\n#{e}")
+            return 0
+        end
+
+        param = mods_conf[/(nbds_max=)(\d+)/]
+        param.split('=').last.to_i
```

Comment:
Make use of pattern matching to get block, i.e. groups (\d+) are returned by the matched object no need to split.last